### PR TITLE
Module resolving and path fixes

### PR DESCRIPTION
**Pull request by schustmi**
## Describe changes

- Makes the source resolving more robust
- Replaces `Path(...).parent` with `os.path.dirname(...)` when copying a directory. The previous solution messed with our path protocols, e.g. it converted `s3://some/directory` to `s3:/some` (missing one slash)
- Sanitizes (replace backslash path separators with forward slashes) all remote paths that our artifact stores get called with
- Only tries to resolve the airflow DAG filepath if not running in a notebook
- Patches a tfx function which crashed when it gets called with one of our remote artifact stores
- Enables copying the global configuration from a windows host to a docker image

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/features/integrations) table.
- [ ] I have added tests to cover my changes.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Comment by AlexejPenner** [comment]
> Can you tell me how to test this on windows? A script would help

Try running this small pipeline using an S3 bucket as Artifact Store:

import numpy as np
import tensorflow as tf

from zenml.integrations.constants import TENSORFLOW
from zenml.logger import get_logger
from zenml.pipelines import pipeline
from zenml.steps import BaseStepConfig, Output, step

logger = get_logger(__name__)


@step
def importer_mnist() -> Output(
    x_train=np.ndarray, y_train=np.ndarray, x_test=np.ndarray, y_test=np.ndarray
):
    """Download the MNIST data store it as an artifact"""
    (x_train, y_train), (
        x_test,
        y_test,
    ) = tf.keras.datasets.mnist.load_data()
    return x_train, y_train, x_test, y_test


@step
def normalizer(
        x_train: np.ndarray, x_test: np.ndarray
) -> Output(x_train_normed=np.ndarray, x_test_normed=np.ndarray):
    """Normalize the values for all the images so they are between 0 and 1"""
    x_train_normed = x_train / 255.0
    x_test_normed = x_test / 255.0
    return x_train_normed, x_test_normed


class TensorflowTrainerConfig(BaseStepConfig):
    """Trainer params"""

    epochs: int = 1
    lr: float = 0.001


@step(enable_cache=False)
def tf_trainer(
        config: TensorflowTrainerConfig,
        x_train: np.ndarray,
        y_train: np.ndarray,
) -> tf.keras.Model:
    """Train a neural net from scratch to recognize MNIST digits return our
    model or the learner"""
    model = tf.keras.Sequential(
        [
            tf.keras.layers.Flatten(input_shape=(28, 28)),
            tf.keras.layers.Dense(10),
        ]
    )

    model.compile(
        optimizer=tf.keras.optimizers.Adam(config.lr),
        loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
        metrics=["accuracy"],
    )

    model.fit(
        x_train,
        y_train,
        epochs=config.epochs,
    )

    # write model
    return model


@pipeline(
    enable_cache=True, required_integrations=[TENSORFLOW]
)
def continuous_deployment_pipeline(
        importer,
        normalizer,
        trainer
):
    # Link all the steps artifacts together
    x_train, y_train, x_test, y_test = importer()
    x_trained_normed, x_test_normed = normalizer(x_train=x_train, x_test=x_test)
    model = trainer(x_train=x_trained_normed, y_train=y_train)


# Initialize a continuous deployment pipeline run
continuous_deployment_pipeline(
    importer=importer_mnist(),
    normalizer=normalizer(),
    trainer=tf_trainer()).run()

**Comment by htahir1** [comment]
trying

**Review by htahir1 — APPROVED** [review]
Ok looks good!

**Review by AlexejPenner — APPROVED** [review]
LGTM

File: src/zenml/config/global_config.py [hunk 1]
```diff
@@ -433,9 +434,14 @@ def copy_active_configuration(
...
-            profile.store_url = store.url.replace(
-                str(config_copy.config_directory), load_config_path
+            store_path = store.get_path_from_url(store.url)
+            assert store_path
+
+            relative_store_path = PurePath(store_path).relative_to(
+                config_copy.config_directory
             )
+            new_store_path = str(load_config_path / relative_store_path)
+            profile.store_url = store.get_local_url(new_store_path)
```
Comment (stefannica) [inline src/zenml/config/global_config.py:444]:
```suggestion
            store_path = store.get_path_from_url(store.url)
            if store_path:
                relative_store_path = PurePath(store_path).relative_to(
                config_copy.config_directory
            )
                new_store_path = str(load_config_path / relative_store_path)
                profile.store_url = store.get_local_url(new_store_path)
            else:
                profile.store_url = store.url
            
```

This doesn't work if you're using a profile pointing to a ZenServer. You can just use `store.url` unaltered as `profile.store_url`, if `get_path_from_url` returns None here.

**Review by stefannica — CHANGES_REQUESTED** [review]
You missed one tiny corner case, everything else looks great.

File: src/zenml/config/global_config.py (hunk 1 above)
Comment (schustmi) [reply src/zenml/config/global_config.py:444]: Nice catch, thank you!
